apps/api/src/campaign-application/campaign-application.service.spec.ts

```diff
@@ -2,14 +2,57 @@ import { Test, TestingModule } from '@nestjs/testing'
 import { CampaignApplicationService } from './campaign-application.service'
 import { CreateCampaignApplicationDto } from './dto/create-campaign-application.dto'
 import { BadRequestException, HttpStatus } from '@nestjs/common'
-
+import {
+  CampaignApplicationState,
+  CampaignState,
+  CampaignTypeCategory,
+  Currency,
+} from '@prisma/client'
+import { CreateCampaignDto } from '../campaign/dto/create-campaign.dto'
+import { prismaMock, MockPrismaService } from '../prisma/prisma-client.mock'
+import { CampaignService } from '../campaign/campaign.service'
+import { NotificationService } from '../sockets/notifications/notification.service'
+import { VaultService } from '../vault/vault.service'
+import { ConfigService } from '@nestjs/config'
+import { MarketingNotificationsService } from '../notifications/notifications.service'
+import { PersonService } from '../person/person.service'
+import { EmailService } from '../email/email.service'
+import { NotificationsProviderInterface } from '../notifications/providers/notifications.interface.providers'
+import { SendGridNotificationsProvider } from '../notifications/providers/notifications.sendgrid.provider'
+import { NotificationGateway } from '../sockets/notifications/gateway'
+import { TemplateService } from '../email/template.service'
 describe('CampaignApplicationService', () => {
   let service: CampaignApplicationService
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [CampaignApplicationService],
-    }).compile()
+      providers: [
+        {
+          // Use the interface as token
+          provide: NotificationsProviderInterface,
+          // But actually provide the service that implements the interface
+          useClass: SendGridNotificationsProvider,
+        },
+        CampaignService,
```

Comment:
Do we need all these providers? CampaignService? Vault Service.... and all the rest? 
It looks like `CampaignApplicationService` is the only requirement for this PR's scope

Reply:
Keep in mind that all of these would need to be instantiated with all of their dependencies for each test!!! 

Instead use mocking (see other tests or look for autoSpy usage)